app/assets/javascripts/global-bar-init.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
```

Comment:
Unless I'm missing something, this code looks like it's been manually de-obfuscated from the minimised JS in `_global_bar.html`. I doubt that's the case, has this been copied from somewhere? If so, should it also be deleted from that place in the same commit as this one?

Reply:
(sorry if I've read your comment wrong) - this code *is* the JS that was previously in the `global_bar.html` file and it has manually been unminified - there wasn't an existing unminified copy of the code that was up to date. It's still required, but I've moved it this `global-bar-init.js` file so it's more readable and testable.

Reply:
(copied from a reply to Ian's comment below, but provides some context to your question too)

Having had a quick look, I don't think it actually needs to be pulled in earlier, so it's possible we could move this code into `global-bar.js`.

I had a lot of trouble with tests when I did that though, so I'm thinking we can keep this as a first pass and iterate in future. It would be good to get this code merged first, so any urgent changes needed post-election can be done with more confidence.

Reply:
You understood perfectly - I assumed it must have existed elsewhere in unminified form already. Well done for deobfuscating it!